packages/mgt-components/src/components/mgt-picker/mgt-picker.ts

```diff
@@ -0,0 +1,270 @@
+    this.keyName = null;
+    this.isRefreshing = false;
+  }
+
+  /**
+   * Refresh the data
+   *
+   * @param {boolean} [hardRefresh=false]
+   * if false (default), the component will only update if the data changed
+   * if true, the data will be first cleared and reloaded completely
+   * @memberof MgtPicker
+   */
+  public refresh(hardRefresh = false) {
+    this.isRefreshing = true;
+    if (hardRefresh) {
+      this.clearState();
+    }
+    this.requestStateUpdate(hardRefresh);
+  }
+
+  /**
+   * Clears the state of the component
+   *
+   * @protected
+   * @memberof MgtPicker
+   */
+  protected clearState(): void {
+    this.response = null;
+  }
+
+  /**
+   * Invoked on each update to perform rendering the picker. This method must return
+   * a lit-html TemplateResult. Setting properties inside this method will *not*
+   * trigger the element to update.
+   */
+  public render() {
+    if (this.isLoadingState && !this.response) {
+      return this.renderTemplate('loading', null);
+    } else if (this.error) {
+      return this.renderTemplate('error', this.error ? this.error : null);
+    } else if (this.hasTemplate('no-data')) {
+      return this.renderTemplate('no-data', null);
+    }
+
+    return this.response?.length > 0 ? this.renderPicker() : this.renderGet();
+  }
+
+  /**
+   * Render picker.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtPicker
+   */
+  protected renderPicker(): TemplateResult {
+    return html`
+      <fluent-combobox id="combobox" autocomplete="list" placeholder=${this.placeholder}>
+        ${this.response.map(
+          item => html`
+          <fluent-option value=${item.id} @click=${e => this.handleClick(e, item)}> ${
+            item[this.keyName]
+          } </fluent-option>`
+        )}
+      </fluent-combobox>
+     `;
+  }
+
+  /**
+   * Render picker.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtPicker
+   */
+  protected renderGet(): TemplateResult {
+    return html`
+      <mgt-get 
+        resource=${this.resource}
+        version=${this.version} 
+        scopes=${this.scopes} 
+        max-pages=${this.maxPages} 
+        ?cache-enabled=${this.cacheEnabled}
+        ?cache-invalidation-period=${this.cacheInvalidationPeriod}>
+      </mgt-get>`;
+  }
+
+  /**
+   * load state into the component.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtPicker
+   */
+  protected async loadState() {
+    if (!this.response) {
+      let parent = this.renderRoot.querySelector('mgt-get');
+      parent.addEventListener('dataChange', e => this.handleDataChange(e));
+    }
+    this.isRefreshing = false;
+  }
+
+  private handleDataChange(e) {
+    let response = e.detail.response.value;
+    let error = e.detail.error ? e.detail.error : null;
+    this.response = response;
+    this.error = error;
+  }
+
+  private handleClick(e, item) {
+    this.fireCustomEvent('selectionChanged', { data: item });
```

Comment:
The data we pass here is set as the detail property of the CustomeEvent that is emitted. Attaching it to a deeper proper does nothing to help the consumer here
```suggestion
    this.fireCustomEvent('selectionChanged', item);
```